docker/build_and_push.Dockerfile

```diff
@@ -9,91 +9,78 @@
-    VENV_PATH="/opt/pysetup/.venv"
+# Use a Python image with uv pre-installed
+FROM ghcr.io/astral-sh/uv:python3.12-bookworm-slim AS builder
+
+# Install the project into `/app`
+WORKDIR /app
+
+# Enable bytecode compilation
+ENV UV_COMPILE_BYTECODE=1
+
+# Copy from the cache instead of linking since it's a mounted volume
+ENV UV_LINK_MODE=copy
 
 RUN apt-get update \
     && apt-get install --no-install-recommends -y \
     # deps for installing poetry
     curl \
     # deps for building python deps
-    build-essential npm \
+    build-essential \
+    # npm
+    npm \
     # gcc
     gcc \
     && apt-get clean \
     && rm -rf /var/lib/apt/lists/*
 
-RUN --mount=type=cache,target=/root/.cache \
-    curl -sSL https://install.python-poetry.org | python3 -
+RUN --mount=type=cache,target=/root/.cache/uv \
+    --mount=type=bind,source=uv.lock,target=uv.lock \
+    --mount=type=bind,source=README.md,target=README.md \
+    --mount=type=bind,source=pyproject.toml,target=pyproject.toml \
+    --mount=type=bind,source=src/backend/base/README.md,target=src/backend/base/README.md \
+    --mount=type=bind,source=src/backend/base/uv.lock,target=src/backend/base/uv.lock \
+    --mount=type=bind,source=src/backend/base/pyproject.toml,target=src/backend/base/pyproject.toml \
+    uv sync --frozen --no-install-project --no-dev
+
+ADD ./src /app/src
+
+COPY src/frontend /tmp/src/frontend
+WORKDIR /tmp/src/frontend
+RUN --mount=type=cache,target=/root/.npm \
+    npm ci \
+    && npm run build \
+    && cp -r build /app/src/backend/langflow/frontend \
+    && rm -rf /tmp/src/frontend
 
 WORKDIR /app
-COPY pyproject.toml poetry.lock README.md ./
-COPY src/ ./src
-COPY scripts/ ./scripts
-RUN python -m pip install requests --user && cd ./scripts && python update_dependencies.py
-
-# 1. Install the dependencies using the current poetry.lock file to create reproducible builds
-# 2. Do not install dev dependencies
-# 3. Install all the extras to ensure all optionals are installed as well
-# 4. --sync to ensure nothing else is in the environment
-# 5. Build the wheel and install "langflow" package (mainly for version)
-
-# Note: moving to build and installing the wheel will make the docker images not reproducible.
-RUN $POETRY_HOME/bin/poetry lock --no-update \
-      # install current lock file with fixed dependencies versions \
-      # do not install dev dependencies \
-      && $POETRY_HOME/bin/poetry install --without dev --sync -E deploy -E couchbase -E cassio \
-      && $POETRY_HOME/bin/poetry build -f wheel \
-      && $POETRY_HOME/bin/poetry run pip install dist/*.whl
+ADD ./pyproject.toml /app/pyproject.toml
+ADD ./uv.lock /app/uv.lock
 
+
+RUN --mount=type=cache,target=/root/.cache/uv \
+uv sync --frozen --no-dev
+FROM python:3.12.3-slim AS runtime
+
+COPY --from=builder /app/.venv /app/.venv
+# Place executables in the environment at the front of the path
+ENV PATH="/app/.venv/bin:$PATH"
+# Reset the entrypoint, don't invoke `uv`
+ENTRYPOINT []
```

Comment:
What does it mean to "not invoke uv" here?

Reply:
often you use `uv run python bla` to use your venv but in the runtime image there is no `uv` (for slimming reasons) so we just add the venv bin to the path and go

Reply:
Also, I think this is an artifact of a previous version. The idea is that the uv image calls uv as an ENTRYPOINT.